src/sentry/search/snuba/executors.py

```diff
@@ -324,6 +331,32 @@ def query(
                 # When its a simple django-only search, we count_hits like normal
                 return paginator.get_result(limit, cursor, count_hits=count_hits, max_hits=max_hits)
 
+        # This search is specific to Inbox. If we're using inbox sort and only querying
+        # postgres then we can use this sort method. Otherwise if we need to go to Snuba,
+        # fail.
+        if (
```

Comment:
Should this be above the `if not end:` bit..in case we stop sending end?

I feel like we could probably merge these if statements ...and actually make use of the of the quick-out for pure postgres queries with an end..

Reply:
I'll think about the merging thing a bit more... I wanted to keep these separate for now so that I could test the other block in production separately, and see how it performs if we start using it again.